### PR TITLE
New Resource: aws_sagemaker_training_job

### PR DESCRIPTION
**Pull request by ddcprg**
Helps resolving #2493

### CODE REVIEW

File: aws/tags_sagemaker.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
+package aws
```
Comment (ddcprg) [inline aws/tags_sagemaker.go]: This file is a copy of the file in #2478

File: aws/validators.go [hunk 2]
```diff
@@ -664,6 +664,24 @@ func validateS3BucketLifecycleRuleId(v interface{}, k string) (ws []string, erro
 	return
 }
 
+func validateSagemakerName(v interface{}, k string) (ws []string, errors []error) {
```
Comment (ddcprg) [inline aws/validators.go]: This function was copied from the same file in #2478

**Comment by ddcprg** [comment]
The build has failed due to S3 errors in `aws/validators_test.go`

**Comment by ddcprg** [comment]
The build error is unrelated to this PR

**Comment by bcornils** [comment]
looks like we have a check failing. Has anyone had time to assess or review criticality of the failure?

**Comment by ddcprg** [comment]
@bcornils  I'll merge master and build again, the error was unrelated to this change last time I checked

**Comment by ibash** [comment]
🙌

**Comment by bflad** [comment]
Hi @ddcprg 👋 Thanks so much for submitting this and sorry for the delayed review process. We have recently merged in a bunch of the Sagemaker resources surrounding model deployment and now we are ready to review/work through other parts of Sagemaker. Could you please rebase this PR? The shared files currently conflicting should be stable now for this pull request. 🤞

**Comment by ddcprg** [comment]
hi @bflad no problem at all, I thought this one was going to be left out for be a static non-deletable resource in AWS. I'll try to take a look at this today and leave an update here

**Comment by ddcprg** [comment]
@bflad I've brought this branch up to date, feel free to take a look any time, cheers

**Comment by nicholasgcoles** [comment]
@ddcprg do you have any idea if this will get merged anytime soon?

**Comment by ddcprg** [comment]
hey @nicholasgcoles I'm not able to tell, I was expecting some feedback from @bflad

**Comment by teamterraform** [comment]
## Notification of Recent and Upcoming Changes to Contributions

Thank you for this contribution! There have been a few recent development changes that affect this pull request. We apologize for the inconvenience, especially if there have been long review delays up until now. Please note that this is automated message from an unmonitored account. See the [FAQ](https://github.com/terraform-providers/terraform-provider-aws/blob/master/docs/FAQ.md) for additional information on the maintainer team and review prioritization.

If you are unable to complete these updates, please leave a comment for the community and maintainers so someone can potentially continue the work. The maintainers will encourage other contributors to use the existing contribution as the base for additional changes as appropriate. Otherwise, contributions that do not receive updated code or comments from the original contributor may be closed in the future so the maintainers can focus on active items.

For the most up to date information about Terraform AWS Provider development, see the [Contributing Guide](https://github.com/terraform-providers/terraform-provider-aws/blob/master/docs/CONTRIBUTING.md). Additional technical debt changes can be tracked with the [`technical-debt` label on issues](https://github.com/terraform-providers/terraform-provider-aws/issues?q=is%3Aopen+is%3Aissue+label%3Atechnical-debt).

As part of updating a pull request with these changes, the most current unit testing and linting will run. These may report issues that were not previously reported.

### Action Required: Terraform 0.12 Syntax

Reference: https://github.com/terraform-providers/terraform-provider-aws/issues/8950
Reference: https://github.com/terraform-providers/terraform-provider-aws/issues/14417

Version 3 and later of the Terraform AWS Provider, which all existing contributions would potentially be added, only supports Terraform 0.12 and later. Certain syntax elements of Terraform 0.11 and earlier show deprecation warnings during runs with Terraform 0.12. Documentation and test configurations, such as those including deprecated string interpolations (`some_attribute = "${aws_service_thing.example.id}"`) should be updated to the newer syntax (`some_attribute = aws_service_thing.example.id`). Contribution testing will automatically fail on older syntax in the near future. Please see the referenced issues for additional information.

### Action Required: Terraform Plugin SDK Version 2

Reference: https://github.com/terraform-providers/terraform-provider-aws/issues/14551

The Terraform AWS Provider has been upgraded to the latest version of the Terraform Plugin SDK. Generally, most changes to contributions should only involve updating Go import paths in source code files. Please see the referenced issue for additional information.

### Action Required: Removal of website/aws.erb File

Reference: https://github.com/terraform-providers/terraform-provider-aws/issues/14712

Any changes to the `website/aws.erb` file are no longer necessary and should be removed from this contribution to prevent merge issues in the near future when the file is removed from the repository. Please see the referenced issue for additional information.

### Upcoming Change of Git Branch Naming

Reference: https://github.com/terraform-providers/terraform-provider-aws/issues/14292

Development environments will need their upstream Git branch updated from `master` to `main` in the near future. Please see the referenced issue for additional information and scheduling.

### Upcoming Change of GitHub Organization

Reference: https://github.com/terraform-providers/terraform-provider-aws/issues/14715

This repository will be migrating from https://github.com/terraform-providers/terraform-provider-aws to https://github.com/hashicorp/terraform-provider-aws. No practitioner or developer action is anticipated and most GitHub functionality will automatically redirect to the new location. Go import paths including `terraform-providers` can remain for now. Please see the referenced issue for additional information and scheduling.

**Comment by DrFaust92** [comment]
Hey @ddcprg, can you rebase and address V2 changes mentioned above and then ill take a look and review this PR.

**Comment by ddcprg** [comment]
@DrFaust92 I've left the company where this code removed it as collaborator no long ago so I cannot longer make changes to this PR. I'll have to close and open a new one in my own fork. I'd like to confirm whether this is actually going to be merged as I don't have a lot of spare time at the moment, last feedback I got is this was a static resource and should not be part of terraform.

**Comment by DrFaust92** [comment]
Let me look at the API and I'll get back to you. I can fork the original branch an continue of you would like.

**Comment by ddcprg** [comment]
any updates on this one @DrFaust92 ? if we can continue where we left that'd be great

**Comment by DrFaust92** [comment]
Hey @ddcprg if you are willing to continue with this PR ill help with the review.

**Comment by DrFaust92** [comment]
Closing, If this is still relevant please open a new issue with the relevant information on why this is resource is needed to be supported by the provider.

**Pull request closed without merging**